concatenate_flows/concatenate_pdms.py

```diff
@@ -0,0 +1,85 @@
+import argparse
+import csv
+import time
+import re
+from core_data_modules.traced_data import TracedData, Metadata
+from core_data_modules.traced_data.io import TracedDataJsonIO
+
+
+def key_normalisation(td_pdm, string_to_replace, replacement_string):
+    """
+    Used to normlise the keys from different PDMs. Acts to the TracedData
+    Object itself(td_pdm)
+
+    :param td_pdm: Data to normalise the keys of
+    :type td_pdm:TracedData object
```

Comment:
This can be just ':type td_pdm: TracedData'

Reply:
Also I'm not sure if this _is_ the correct type (I realise that Python doesn't type-check for you so this is a bit painful). Given that you iterate over td_pdm and then over the keys of that, I _think_ this should be an `iterable of TracedData`?